### ISSUE TITLE
Song Lookup

### ISSUE THREAD

**Issue by mguida22** [GitSound/GitSound#14]
Enter a song title, maybe artist and get the song id

**Comment by mguida22** [comment]
This could be helpful, specifically the way it finds songs from lyrics. Probably more useful for a later version but whatever. https://github.com/yask123/Instant-Music-Downloader

**Comment by nicolele** [comment]
Example query for "Diamonds From Sierra Leone - Remix - Album Version (Explicit)":

``` JSON
{
  "tracks" : {
    "href" : "https://api.spotify.com/v1/search?query=track%3ADiamonds+From+Sierra+Leone+-+Remix+-+Album+Version+%28Explicit%29&offset=0&limit=10&type=track",
    "items" : [ {
      "album" : {
        "album_type" : "album",
        "available_markets" : [ "CA", "MX", "US" ],
        "external_urls" : {
          "spotify" : "https://open.spotify.com/album/5ll74bqtkcXlKE7wwkMq4g"
        },
        "href" : "https://api.spotify.com/v1/albums/5ll74bqtkcXlKE7wwkMq4g",
        "id" : "5ll74bqtkcXlKE7wwkMq4g",
        "images" : [ {
          "height" : 635,
          "url" : "https://i.scdn.co/image/901a0af6765c05a59e8bd4d203945e340628a9dd",
          "width" : 640
        }, {
          "height" : 298,
          "url" : "https://i.scdn.co/image/1a2edf19369a111a655897272e7ea9a2d909c0ef",
          "width" : 300
        }, {
          "height" : 64,
          "url" : "https://i.scdn.co/image/0f9a22c40789f19ae41b7a606e3516b447eb514c",
          "width" : 64
        } ],
        "name" : "Late Registration (Explicit Version)",
        "type" : "album",
        "uri" : "spotify:album:5ll74bqtkcXlKE7wwkMq4g"
      },
      "artists" : [ {
        "external_urls" : {
          "spotify" : "https://open.spotify.com/artist/5K4W6rqBFWDnAN6FQUkS6x"
        },
        "href" : "https://api.spotify.com/v1/artists/5K4W6rqBFWDnAN6FQUkS6x",
        "id" : "5K4W6rqBFWDnAN6FQUkS6x",
        "name" : "Kanye West",
        "type" : "artist",
        "uri" : "spotify:artist:5K4W6rqBFWDnAN6FQUkS6x"
      }, {
        "external_urls" : {
          "spotify" : "https://open.spotify.com/artist/3nFkdlSjzX9mRTtwJOzDYB"
        },
        "href" : "https://api.spotify.com/v1/artists/3nFkdlSjzX9mRTtwJOzDYB",
        "id" : "3nFkdlSjzX9mRTtwJOzDYB",
        "name" : "JAY Z",
        "type" : "artist",
        "uri" : "spotify:artist:3nFkdlSjzX9mRTtwJOzDYB"
      } ],
      "available_markets" : [ "CA", "MX", "US" ],
      "disc_number" : 1,
      "duration_ms" : 233400,
      "explicit" : true,
      "external_ids" : {
        "isrc" : "USUM70502927"
      },
      "external_urls" : {
        "spotify" : "https://open.spotify.com/track/34KUIBsIUiPV7oCIzSdDAU"
      },
      "href" : "https://api.spotify.com/v1/tracks/34KUIBsIUiPV7oCIzSdDAU",
      "id" : "34KUIBsIUiPV7oCIzSdDAU",
      "name" : "Diamonds From Sierra Leone - Remix - Album Version (Explicit)",
      "popularity" : 58,
      "preview_url" : "https://p.scdn.co/mp3-preview/f13b19f368bd127ab36ad00ce785cfa97c9e2dd2",
      "track_number" : 13,
      "type" : "track",
      "uri" : "spotify:track:34KUIBsIUiPV7oCIzSdDAU"
    }, {
      "album" : {
        "album_type" : "album",
        "available_markets" : [ "AD", "AR", "AT", "AU", "BE", "BG", "BO", "BR", "CH", "CL", "CO", "CR", "CY", "CZ", "DE", "DK", "DO", "EC", "EE", "ES", "FI", "FR", "GR", "GT", "HK", "HN", "HU", "IS", "IT", "LI", "LT", "LU", "LV", "MC", "MT", "MY", "NI", "NL", "NO", "NZ", "PA", "PE", "PH", "PL", "PT", "PY", "RO", "SE", "SG", "SI", "SK", "SV", "TR", "TW", "UY" ],
        "external_urls" : {
          "spotify" : "https://open.spotify.com/album/4GRDFQ9HRoO0by8H0r2a3I"
        },
        "href" : "https://api.spotify.com/v1/albums/4GRDFQ9HRoO0by8H0r2a3I",
        "id" : "4GRDFQ9HRoO0by8H0r2a3I",
        "images" : [ {
          "height" : 635,
          "url" : "https://i.scdn.co/image/2c27422325edc0d54522a0aadd92329c565dc618",
          "width" : 640
        }, {
          "height" : 298,
          "url" : "https://i.scdn.co/image/d1fa0855e8978f8d33b93bfa16f3c0b25ea08daa",
          "width" : 300
        }, {
          "height" : 64,
          "url" : "https://i.scdn.co/image/a5fa804a6b17d2b76d89c757a397a1d586357cf7",
          "width" : 64
        } ],
        "name" : "Late Registration (Explicit Version)",
        "type" : "album",
        "uri" : "spotify:album:4GRDFQ9HRoO0by8H0r2a3I"
      },
      "artists" : [ {
        "external_urls" : {
          "spotify" : "https://open.spotify.com/artist/5K4W6rqBFWDnAN6FQUkS6x"
        },
        "href" : "https://api.spotify.com/v1/artists/5K4W6rqBFWDnAN6FQUkS6x",
        "id" : "5K4W6rqBFWDnAN6FQUkS6x",
        "name" : "Kanye West",
        "type" : "artist",
        "uri" : "spotify:artist:5K4W6rqBFWDnAN6FQUkS6x"
      }, {
        "external_urls" : {
          "spotify" : "https://open.spotify.com/artist/3nFkdlSjzX9mRTtwJOzDYB"
        },
        "href" : "https://api.spotify.com/v1/artists/3nFkdlSjzX9mRTtwJOzDYB",
        "id" : "3nFkdlSjzX9mRTtwJOzDYB",
        "name" : "JAY Z",
        "type" : "artist",
        "uri" : "spotify:artist:3nFkdlSjzX9mRTtwJOzDYB"
      } ],
      "available_markets" : [ "AD", "AR", "AT", "AU", "BE", "BG", "BO", "BR", "CH", "CL", "CO", "CR", "CY", "CZ", "DE", "DK", "DO", "EC", "EE", "ES", "FI", "FR", "GR", "GT", "HK", "HN", "HU", "IS", "IT", "LI", "LT", "LU", "LV", "MC", "MT", "MY", "NI", "NL", "NO", "NZ", "PA", "PE", "PH", "PL", "PT", "PY", "RO", "SE", "SG", "SI", "SK", "SV", "TR", "TW", "UY" ],
      "disc_number" : 1,
      "duration_ms" : 233400,
      "explicit" : true,
      "external_ids" : {
        "isrc" : "USUM70502927"
      },
      "external_urls" : {
        "spotify" : "https://open.spotify.com/track/2d5NhldpaetRQBVbNveDkK"
      },
      "href" : "https://api.spotify.com/v1/tracks/2d5NhldpaetRQBVbNveDkK",
      "id" : "2d5NhldpaetRQBVbNveDkK",
      "name" : "Diamonds From Sierra Leone - Remix - Album Version (Explicit)",
      "popularity" : 48,
      "preview_url" : "https://p.scdn.co/mp3-preview/e2d4d350b35d8eca22874a556a4cf1d7507664c8",
      "track_number" : 13,
      "type" : "track",
      "uri" : "spotify:track:2d5NhldpaetRQBVbNveDkK"
    }, {
      "album" : {
        "album_type" : "album",
        "available_markets" : [ "GB", "IE" ],
        "external_urls" : {
          "spotify" : "https://open.spotify.com/album/0Ds6i3h0F9RcYIKAD5Olum"
        },
        "href" : "https://api.spotify.com/v1/albums/0Ds6i3h0F9RcYIKAD5Olum",
        "id" : "0Ds6i3h0F9RcYIKAD5Olum",
        "images" : [ {
          "height" : 634,
          "url" : "https://i.scdn.co/image/7f8692b6b0bc81a062ac11547d0aa7af3ace328f",
          "width" : 640
        }, {
          "height" : 297,
          "url" : "https://i.scdn.co/image/c8679b191c5e9207e99b1af7d5341ae88b48851e",
          "width" : 300
        }, {
          "height" : 63,
          "url" : "https://i.scdn.co/image/ddb9c9138f7006170de0e4a23b54a710214a4f7d",
          "width" : 64
        } ],
        "name" : "Late Registration",
        "type" : "album",
        "uri" : "spotify:album:0Ds6i3h0F9RcYIKAD5Olum"
      },
      "artists" : [ {
        "external_urls" : {
          "spotify" : "https://open.spotify.com/artist/5K4W6rqBFWDnAN6FQUkS6x"
        },
        "href" : "https://api.spotify.com/v1/artists/5K4W6rqBFWDnAN6FQUkS6x",
        "id" : "5K4W6rqBFWDnAN6FQUkS6x",
        "name" : "Kanye West",
        "type" : "artist",
        "uri" : "spotify:artist:5K4W6rqBFWDnAN6FQUkS6x"
      }, {
        "external_urls" : {
          "spotify" : "https://open.spotify.com/artist/3nFkdlSjzX9mRTtwJOzDYB"
        },
        "href" : "https://api.spotify.com/v1/artists/3nFkdlSjzX9mRTtwJOzDYB",
        "id" : "3nFkdlSjzX9mRTtwJOzDYB",
        "name" : "JAY Z",
        "type" : "artist",
        "uri" : "spotify:artist:3nFkdlSjzX9mRTtwJOzDYB"
      } ],
      "available_markets" : [ "GB", "IE" ],
      "disc_number" : 1,
      "duration_ms" : 233400,
      "explicit" : true,
      "external_ids" : {
        "isrc" : "USUM70502927"
      },
      "external_urls" : {
        "spotify" : "https://open.spotify.com/track/4ZLNj4Zlp6XZ3WAVpPQEnq"
      },
      "href" : "https://api.spotify.com/v1/tracks/4ZLNj4Zlp6XZ3WAVpPQEnq",
      "id" : "4ZLNj4Zlp6XZ3WAVpPQEnq",
      "name" : "Diamonds From Sierra Leone - Remix - Album Version (Explicit)",
      "popularity" : 44,
      "preview_url" : "https://p.scdn.co/mp3-preview/ef404fd3fd255d35e5a5c38fa2a8c5db146faea6",
      "track_number" : 13,
      "type" : "track",
      "uri" : "spotify:track:4ZLNj4Zlp6XZ3WAVpPQEnq"
    } ],
    "limit" : 10,
    "next" : null,
    "offset" : 0,
    "previous" : null,
    "total" : 3
  }
}
```

**Comment by mguida22** [comment]
Closed by #19
